ape_alchemy/provider.py

```diff
@@ -132,23 +139,45 @@ def get_virtual_machine_error(self, exception: Exception) -> VirtualMachineError
 
         return VirtualMachineError(message=message)
 
-    def _make_request(self, endpoint: str, parameters: list) -> Any:
-        try:
-            return super()._make_request(endpoint, parameters)
-        except HTTPError as err:
-            response_data = err.response.json()
-            if "error" not in response_data:
-                raise AlchemyProviderError(str(err)) from err
-
-            error_data = response_data["error"]
-            message = (
-                error_data.get("message", str(error_data))
-                if isinstance(error_data, dict)
-                else error_data
-            )
-            cls = (
-                AlchemyFeatureNotAvailable
-                if "is not available" in message
-                else AlchemyProviderError
-            )
-            raise cls(message) from err
+    def _make_request(
+        self,
+        endpoint: str,
+        parameters: list,
+        min_retry_delay: int=1_000, # 1 second
+        retry_backoff_factor: int=2, # exponential backoff
```

Comment:
I think it is good to have these as kwargs, but it might also be nice to have them as properties you can set in the `ape-config` file.

The properties I mean are `min_retry_delay`, `retry_backoff_factor`, `max_retry_delay`, `max_retries`, and `retry_jitter`.

There currently is not a config class registered for this plugin, but I promise it's easy!
First, define the class at the top of the module:

```python
from ape.api import PluginConfig

class AlchemyConfig(PluginConfig):
   min_retry_delay: int = 1_000
   """
   The amount of milliseconds to wait before retrying the request.
   Defaults to one second.
   """
```

^ Put all of the same defaults you are using here in the config.

Then, in `__init__.py`, register it like this:

```python
from .provider import AlchemyConfig

@plugins.register(plugins.Config)
def config_class():
    yield AlchemyConfig
```

Then, in this method, all of these params can be `Optional[int]` and default to `None`. And you can grab each value like , taking in account of possible config values:

```python
from typing import cast

alchemy_config = cast(AlchemyConfig, self.config_manager.get_config("alchemy"))
retry_backoff_factor = retry_backoff_factor if retry_backoff_factor is not None else alchemy_config.retry_backoff_factor
...
```